From 120c1d7c18f60f91acaa0fcd7708cdac35c72481 Mon Sep 17 00:00:00 2001
From: Prabhu Sundararaj <prabhu.sundararaj@nxp.com>
Date: Thu, 25 May 2017 15:17:06 -0500
Subject: [PATCH] MGS-2921[#ccc] Fix gl-renderer of using
 create_platform_window

create_platform_window must be used only when get_platform_display
API used. Some cases when NO_EGL_PLATFORM is used, the display uses
legacy method and create EGL surface use platform method causing crash.

Date: May 25, 2017
Signed-off-by: Prabhu Sundararaj <prabhu.sundararaj@nxp.com>
---
 src/gl-renderer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gl-renderer.c b/src/gl-renderer.c
index 23c0cd7..ed4ecdb 100644
--- a/src/gl-renderer.c
+++ b/src/gl-renderer.c
@@ -2578,7 +2578,7 @@ gl_renderer_output_create(struct weston_output *output,
 	if (go == NULL)
 		return -1;
 
-	if (gr->create_platform_window) {
+	if (gr->create_platform_window && get_platform_display) {
 		go->egl_surface =
 			gr->create_platform_window(gr->egl_display,
 						   egl_config,
-- 
2.7.4

